src/lib/core/components/visualizations/implementations/ScatterplotVisualization.tsx

```diff
@@ -588,8 +588,9 @@ function ScatterplotViz(props: VisualizationProps<Options>) {
       let gradientColorscaleType: string | undefined;
 
       if (
-        overlayVariable?.type === 'integer' ||
-        overlayVariable?.type === 'number'
+        'seriesGradientColorscale' in response.scatterplot.data[0] &&
+        (overlayVariable?.type === 'integer' ||
+          overlayVariable?.type === 'number')
```

Comment:
Is checking the first series enough/correct, or should I check all series?

Reply:
I'm guessing it would be a bug if they weren't all the same, in this regard. If we have some assurance of upstream of this code, then I think only checking the first is fine.

That being said, if `response.scatterplot.data` is an empty array, a runtime error will be thrown here.

The safest thing might be to do something like this:

```typescript
response.scatterplot.data.every(t => 'seriesGradientColorscale' in t)
```

Reply:
That sounds like a plan from Dave.

My deep contribution is to add a blank line after this `if {}` block just before
```
      const overlayVocabulary = computedOverlayVariableDescriptor
```

Reply:
I agree with both suggestions. Thanks y'all! :)